modules/nw-egress-ips-about.adoc

```diff
@@ -1,7 +1,101 @@
 // Module included in the following assemblies:
 //
-// * networking/assigning-egress-ips.adoc
+// * networking/openshift_sdn/assigning-egress-ips.adoc
+// * networking/ovn_kubernetes_network_provider/assigning-egress-ips.adoc
 
+ifeval::["{context}" == "egress-ips"]
+:openshift-sdn:
+endif::[]
+ifeval::["{context}" == "ovn-kubernetes-configuring-egress-ips"]
+:ovn:
+endif::[]
+
+ifdef::ovn[]
+[id="nw-egress-ips-about_{context}"]
+= Egress IP address architectural design and implementation
+
+The {product-title} egress IP address functionality allows you to ensure that the traffic from one or more pods in one or more namespaces have a consistent source IP address for services outside the cluster network.
+
+For example, you might have a pod that periodically queries a database hosted on a server outside of your cluster. To enforce access requirements for the server, a packet filtering device is configured to allow traffic only from specific IP addresses.
+You can configure a specific egress IP address for the pod making the requests to the server, to ensure that you can reliably allow access to the server from only that specific pod.
+
+An egress IP address is implemented as an additional IP address on the primary network interface of a node and must be in the same subnet as the primary IP address of the node.
+
+[id="nw-egress-ips-considerations_{context}"]
+== Assignment of egress IPs to pods
+
+To assign one or more egress IPs to a namespace or specific pods in a namespace, the following conditions must be satisfied:
+
+- You must have at least one node in your cluster with the `k8s.ovn.org/egress-assignable: ""` label.
+- You must create an `EgressIP` object specifying one or more egress IP addresses to use as the source IP address for traffic leaving the cluster from pods in a namespace.
+
+[IMPORTANT]
+====
+If you create `EgressIP` objects prior to labeling any nodes in your cluster for egress IP assignment, {product-title} might assign every egress IP address to the first node with the `k8s.ovn.org/egress-assignable: ""` label.
```

Comment:
@alexanderConstantinescu, can I elaborate on this? The enhancement doesn't seem to provide any additional insight into why to avoid this:

> Points 4.1. - 4.6. means: that if a user creates `EgressIP` objects **before** tagging nodes, then there is a risk that most `EgressIP` objects will be assigned to the same egress node, should that node be able to host them. It would be impossible for OVN-Kubernetes to rectify that situation, as it cannot know how many egress nodes will be available to it, so it will try to assign everything in its queue to  the first node it sees. It is thus up to the user to delete and re-create all objects should a lot of them have been created that way.
It is subsequently strongly advised that a user first labels all nodes needed, and then creates `EgressIP` objects.

Thanks!

Reply:
The reason why someone would like to avoid that is: as to not have all egress IPs / majority of then assigned to one node. It would be better if all egress IPs are evenly distributed (or as much so as possible) across all egress nodes across the cluster. And that even distribution can only be made once OVN-Kubernetes knows all egress nodes _prior_ to making each egress IP assignment.